jsonschema/validators.py

```diff
@@ -495,10 +585,7 @@ def __init__(
         self.handlers = dict(handlers)
 
         self._scopes_stack = [base_uri]
-        self.store = _utils.URIDict(
-            (id, validator.META_SCHEMA)
-            for id, validator in meta_schemas.items()
-        )
+        self.store = get_store()
```

Comment:
Why are we turning what was previously local state to mutable global state?

Reply:
Since `draft2020-12` relies on references in takes some time to initialize the validator as it fetches all this URLs form remote. The tests for `draft2020-12` require over 30 min without this approach and takes several hours with the current pipeline configuration as there is a limit on paralel github actions.

Reply:
That sounds like there are some URLs missing from Draft202012Validator's own schema store.

IIRC from the last time I did this myself, basically we need to be caching the vocabulary schemas as well (globally), not just the metaschema.

But yeah we shouldn't move to global state here for all URLs.

Reply:
Yes, the meta schema is loaded form the local file located in `draft2020-12.json`.  All references are loaded afterwards:

```
{
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "$id": "https://json-schema.org/draft/2020-12/schema",
    ...
    "allOf": [
        {"$ref": "meta/core"},
        {"$ref": "meta/applicator"},
        {"$ref": "meta/unevaluated"},
        {"$ref": "meta/validation"},
        {"$ref": "meta/meta-data"},
        {"$ref": "meta/format-annotation"},
        {"$ref": "meta/content"}
    ],
    ...
}
```

In this case I will try to load only the meta schemas in a global store, everything else will be loaded in the validator store.

Reply:
Now the vocabulary schemas are also stored in the `schemas` directory and loaded additionally to the store by adding `vocabulary_schemas` when creating a validator.